gnd/src/main/java/com/google/android/gnd/persistence/geojson/GeoJsonParser.java

```diff
@@ -89,19 +80,39 @@
           .map(this::jsonToTile)
           .collect(toImmutableList());
 
-    } catch (IOException | JSONException e) {
-      Log.e(TAG, "Unable to load JSON layer", e);
+    } catch (JSONException e) {
+      Log.e(TAG, "Unable to parse JSON", e);
     }
+
     return ImmutableList.of();
   }
 
+  /**
+   * Converts a JSONArray to an array of JSONObjects. Provided for compatibility with java8 streams.
+   * JSONArray itself only inherits from Object, and is not convertible to a stream.
+   */
+  private static JSONObject[] toArray(JSONArray arr) {
+    JSONObject[] result = new JSONObject[arr.length()];
+
+    for (int i = 0; i < arr.length(); i++) {
+      try {
+        JSONObject o = arr.getJSONObject(i);
+        result[i] = o;
+      } catch (JSONException e) {
+        Log.e(TAG, "couldn't parse json", e);
+      }
+    }
+
+    return result;
+  }
+
   /** Returns the {@link Tile} specified by {@param json}. */
   private Tile jsonToTile(GeoJsonTile json) {
     return Tile.newBuilder()
         .setId(uuidGenerator.generateUuid())
-        .setUrl(json.getUrl().get())
+        .setUrl(json.getUrl().orElse(""))
         .setState(State.PENDING)
-        .setPath(Tile.pathFromId(json.getId().get()))
+        .setPath(Tile.pathFromId(json.getId().orElse("")))
```

Comment:
Ideally we wouldn't keep Tile in an invalidate state if the file is invalid. Could you add a TODO here to throw an Exception instead that gets handled downstream?

Reply:
Added.